### PR TITLE
fix JitConfig initialize and destoy

### PR DESCRIPTION
**Pull request by sandreenko**
We don't use "destroy" nowadays, but if we want, it will not work.
It is part of spmi collection with different environment sets.

### CODE REVIEW

**Comment by jkotas** [comment]
@dotnet-bot test this please

File: src/jit/jitconfig.cpp [hunk 1]
```diff
@@ -45,7 +45,7 @@ void JitConfigValues::MethodSet::initialize(const wchar_t* list, ICorJitHost* ho
     {
         // Failed to convert the list. Free the memory and ignore the list.
         host->freeMemory(reinterpret_cast<void*>(const_cast<char*>(m_list)));
-        m_list = "";
+        m_list = nullptr;
```
Comment (AndyAyersMS) [inline src/jit/jitconfig.cpp]: These failure cases are probably rare, but is downstream code prepared for `m_list` to be null?

If not you might add a new field to indicate that `m_list` should not be freed, and set that here and keep the value as the empty string, then check the new field in the destructor.

Comment (sandreenko) [inline src/jit/jitconfig.cpp]: It is often, it can't convert if utf8ListLen == 0, that is empty config string case.

**Comment by sandreenko** [comment]
@AndyAyersMS I checked that we have all necessary checks. We access it only through pointers to method names (like `m_list[name->m_methodNameStart]`), but if `m_list == null`, then `m_names== null`.

**Comment by AndyAyersMS** [comment]
I don't see where `m_names` is set to null initially, but I suppose this happens because the only instances of this type are static objects. I would recommend asserting that `m_names` is nullptr in `initialize` just like we do for `m_list` in case anyone ever stack or heap allocates on of these.

Also there are uses of the `list()` method, though it appears consumers there are set up tolerate getting null back.

**Comment by sandreenko** [comment]
The assert was added.

**Pull request merged**